src/hipscat/pixel_tree/pixel_alignment.py

```diff
@@ -131,6 +134,70 @@ def get_pixel_mapping_df(mapping: np.ndarray, map_order: int) -> pd.DataFrame:
     return result_mapping
 
 
+# pylint: disable=too-many-statements
+@njit(numba.int64[::1, :](numba.int64[:, :], numba.int64[:, :]))
+def perform_inner_align_trees(
+    left: np.ndarray,
+    right: np.ndarray,
+) -> np.ndarray:
+    """Performs an inner alignment on arrays of pixel intervals
+
+    Pixel interval lists must be of to the same order
+
+    Args:
+        left (np.ndarray): the left array of intervals
+        right (np.ndarray): the right array of intervals
+
+    Returns (List[np.ndarray]):
+        The pixel mapping of the matching left, right, and aligned pixels with each row containing an array of
+        [left_order, left_pixel, right_order, right_pixel, aligned_order, aligned_pixel]
+    """
+    max_out_len = left.shape[0] + right.shape[0]
+    mapping = np.zeros((max_out_len, 6), dtype=np.int64)
+    left_index = 0
+    right_index = 0
+    out_len = 0
```

Comment:
(Optional): It might be a little easier to understand the interior of the code if `out_len` was called `out_index`.

Reply:
I changed it to `last_index`, I think that's most descriptive

Reply:
no, actually I think you're right and `out_index` is better